python/cudf/cudf/tests/test_sorting.py

```diff
@@ -269,7 +269,7 @@ def test_dataframe_scatter_by_map(map_size, nelem, keep):
     df["a"] = np.random.choice(strlist[:map_size], nelem)
     df["b"] = np.random.uniform(low=0, high=map_size, size=nelem)
     df["c"] = np.random.randint(map_size, size=nelem)
-    df["d"] = df["a"]._column.as_categorical_column(np.int32)
+    df["d"] = np.zeros(nelem, dtype=np.int32)
```

Comment:
Isn't it possible to keep a categorical column?

Reply:
This was previously creating a column of type `int32`. As having non-numerics used to give 0's in the result when there was a type-cast and now the same scenario will error, changed the column to directly initialize with 0's.

Reply:
Earlier the test had a categorical column and now it has numerical column, so we are missing to test categorical. You can create a categorical column with only integers to test this.

Reply:
Added the categorical back 👍